apps/web/src/lib/data/window/preset-storage.ts

```diff
@@ -0,0 +1,170 @@
+/**
+ * @license BSD-3-Clause
+ * Copyright (c) 2023, ッツ Reader Authors
+ * All rights reserved.
+ */
+
+import { Subject, BehaviorSubject } from 'rxjs';
+import { localStorage } from './local-storage';
+
+export class PresetStorage {
+  private _preset: string;
+
+  private bookPresets$: BehaviorSubject<Record<string, string>>; // idk how better to avoid cyclic dependency
+
+  public presetChanged: Subject<undefined>;
+
+  /* eslint-disable-next-line @typescript-eslint/default-param-last */
+  constructor(preset: string = 'Global', bookPresets$: BehaviorSubject<Record<string, string>>) {
+    this._preset = preset;
+    this.bookPresets$ = bookPresets$;
+    this.presetChanged = new Subject<undefined>();
+  }
+
+  setItem(key: string, value: string) {
+    const allPresetSettingsValue = localStorage.getItem('presetSettings') ?? '{}';
+    let allPresetSettingsObj;
+    let presetSettings;
+
+    try {
+      allPresetSettingsObj = JSON.parse(allPresetSettingsValue);
+      presetSettings = allPresetSettingsObj[this._preset] ?? {};
+    } catch (error) {
+      allPresetSettingsObj = {};
+      presetSettings = {};
+    }
+
+    presetSettings[key] = value;
+    allPresetSettingsObj[this._preset] = presetSettings;
+    localStorage.setItem('presetSettings', JSON.stringify(allPresetSettingsObj));
+  }
+
+  getItem(key: string) {
```

Comment:
returning undefined would bring the issue that it would fallback to the default value - everything the user has adjusted would be ignored.. it needs to be current preset > localstorage value > default value